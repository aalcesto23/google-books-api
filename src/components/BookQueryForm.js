import React, {Component} from 'react'
import '../'
class BookQueryForm extends Component {
  render() {
    return (
      <form onSubmit={this.props.getBooks}>
        <input className='input-field col 12' type='text' name='query' onChange={this.props.updateQuery} placeholder='Search...' />
        <button className="waves-effect waves-light btn-large" disabled={this.props.query === ""} type="submit" name="action"><i className="material-icons right">search</i>SEARCH</button>
      </form>
    )
  }
}

export default BookQueryForm;