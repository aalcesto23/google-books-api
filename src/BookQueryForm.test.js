import React from 'react';
import ReactDOM from 'react-dom';
import BookQueryForm from './components/BookQueryForm';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

it('Form renders', () => {
  const queryForm = renderer.create(
    <BookQueryForm />
  )
  let form = queryForm.toJSON();
  expect(form).toMatchSnapshot();
});

it('Button is disabled if query is empty', () => {
  const form = shallow(<BookQueryForm query="" />);
  const button = form.find('button')
  expect(button.props().disabled).toBe(true)
})

it('Button is enabled if query has content', () => {
  const form = shallow(<BookQueryForm query="Harry Potter" />);
  const button = form.find('button')
  expect(button.props().disabled).toBe(false)
})