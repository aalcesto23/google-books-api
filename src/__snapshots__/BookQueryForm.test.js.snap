// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `
<form>
  <input
    className="input-field col 12"
    name="query"
    placeholder="Search..."
    type="text"
  />
  <button
    className="waves-effect waves-light btn-large"
    disabled={false}
    name="action"
    type="submit"
  >
    <i
      className="material-icons right"
    >
      search
    </i>
    SEARCH
  </button>
</form>
`;
